ui/src/main/java/org/jtaccuino/core/ui/JavaCellFactory.java

```diff
@@ -206,13 +212,29 @@ private JavaCellSkin(JavaCell javaCell) {
                         }
                     });
 
+            var outputNodes = javaCell.getCellData().getOutputData().stream().map(od
+                    -> od.mimeBundle().entrySet().stream().map(me
+                            -> switch (me.getKey()) {
+                case "image/png" ->
+                    new ImageView(new Image(new ByteArrayInputStream(Base64.getDecoder().decode(me.getValue()))));
+                case "text/plain" ->
+                    new Label(me.getValue());
+                default ->
+                    new Label(me.getValue());
+            }
+                    ).toList()
+            ).toList();
+
+            outputNodes.forEach(nodes -> nodes.forEach(n -> outputBox.getChildren().add(n)));
```

Comment:
Streaming over a collection of a map which is turned into a two-level list.
If you flatMap the streaming of the entrySets you'll only have a single level of collection

```suggestion
            var outputNodes = javaCell.getCellData().getOutputData().stream()
                    .flatMap(od -> od.mimeBundle().entrySet().stream())
                    .map(me
                            -> switch (me.getKey()) {
                case "image/png" ->
                    new ImageView(new Image(new ByteArrayInputStream(Base64.getDecoder().decode(me.getValue()))));
                case "text/plain" ->
                    new Label(me.getValue());
                default ->
                    new Label(me.getValue());
            }
            ).toList();

            outputNodes.forEach(outputBox.getChildren()::add);
```